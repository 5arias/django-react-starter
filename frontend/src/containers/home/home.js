import React, { Component } from 'react'
import { connect } from 'react-redux'
import { sampleAction } from '../../redux/actions'
import logo from './logo.svg'
import css from './home.css'

const mapStateToProps = state => {
  return {
    content: state.sample.sampleKey
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateContent: text => dispatch(sampleAction(text))
  }
}

export class Home extends Component {
  componentDidMount() {
    this.props.updateContent(
      'To get started, edit this content in src/containers/home.'
    )
  }

  render() {
    return (
      <React.Fragment>
        <header className={css.header}>
          <img src={logo} className={css.logo} alt="logo" />
          <h1 className={css.title}>Welcome to R3CSS</h1>
        </header>
        <p className={css.intro}>{this.props.content}</p>
      </React.Fragment>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
