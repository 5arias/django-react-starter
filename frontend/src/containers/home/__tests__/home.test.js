import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Home } from '../home'

const setup = propOverrides => {
  const props = Object.assign(
    {
      content: 'this is the home content',
      updateContent: jest.fn()
    },
    propOverrides
  )

  const wrapper = shallow(<Home {...props} />)

  return {
    props,
    wrapper
  }
}

describe('Home', () => {
  it('calls updateContent when mounted', () => {
    const { props, wrapper } = setup()
    expect(props.updateContent).toHaveBeenCalled()
  })

  it('renders and matches the snapshot', () => {
    const { wrapper } = setup()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
