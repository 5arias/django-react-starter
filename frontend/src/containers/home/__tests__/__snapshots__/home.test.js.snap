// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders and matches the snapshot 1`] = `
<React.Fragment>
  <header>
    <img
      alt="logo"
      src="logo.svg"
    />
    <h1>
      Welcome to R3CSS
    </h1>
  </header>
  <p>
    this is the home content
  </p>
</React.Fragment>
`;
